Classes/Controller/ContentController.php

```diff
@@ -52,6 +55,27 @@ public function renderAction()
             );
         $flexform->setTyposcriptDefaults($this->settings['defaults']);
 
+        if (empty(str_replace("\r\n", '', $contentElement['bodytext']))) {
+            $file = $flexform->getCFile();
+            $linkService = GeneralUtility::makeInstance(LinkService::class);
+            $data = $linkService->resolveByStringRepresentation($file);
```

Comment:
Nice idea to just reuse the `LinKService` here in order to decide whether it's an URL or link.